branchpro/_sliders.py

```diff
@@ -0,0 +1,72 @@
+#
+# Sliders Class
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import dash_core_components as dcc
+import dash_html_components as html
+
+import numpy as np
+
+
+class _SliderComponent():
+    """_SliderComponent Class
+    Stores slider information for sliders in branchpro Dash app.
+    """
+    def __init__(self):
+        self._sliders = []
+        self._slider_ids = []
+
+    def add_slider(self, label, new_id, init_val, min_val, max_val, step_size):
+        """
+        Creates a new slider with label for the Dash app plot.
+
+        Parameters
+        ----------
+        label
+            Title text shown above slider.
+        new_id
+            ID of slider (internal, not shown in app).
+        init_val
+            Initial slider value (default position).
+        min_val
+            Minimum (leftmost) slider value.
+        max_val
+            Maximum (rightmost) slider value.
+        step_size
+            Incremement between slider values.
+        """
+        new_slider = [
+                        html.Label(label),
+                        dcc.Slider(
+                            id=new_id,
+                            min=min_val,
+                            max=max_val,
+                            value=init_val,
+                            marks={str(ri): str(ri) for ri in np.arange(
```

Comment:
This is causing problems---for small step sizes, there are far too many marks and they all overlap.
One option might be to choose a fixed reasonable number of marks upfront, or see if there is some default already implemented.
Alternatively, we could have marks only at the start and end, and find a way to display the current value of the slider as you move it.

Reply:
I'll see what I can do about that.

Reply:
Add requirement for marks list to have maximum 11 elements (10 intervals shown)